### PR TITLE
Add cypress test for workspace overview page

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description

Add cypress test for essential/search/analytics overview page

image

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_search_overviews.spec.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        permissions: {
+          library_write: { users: ['%me%'] },
+          write: { users: ['%me%'] },
+        },
+        ...(datasourceId ? { dataSources: [datasourceId] } : {}),
+      },
+    }).then((value) => {
+      workspaceId = value;
+    });
+  };
```
Comment (SuZhou-Joe) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_search_overviews.spec.js:35]:
```suggestion
  const createWorkspace = (datasourceId) => {
    cy.createWorkspace({
      name: workspaceName,
      description: workspaceDescription,
      features: workspaceFeatures,
      settings: {
        permissions: {
          library_write: { users: ['%me%'] },
          write: { users: ['%me%'] },
        },
        ...(datasourceId ? { dataSources: [datasourceId] } : {}),
      },
    }).then((value) => {
      workspaceId = value;
    });
  };
```
We can move the function to a util function maybe?

File: cypress/utils/commands.js [hunk 2]
```diff
@@ -595,6 +595,28 @@ Cypress.Commands.add('loadSampleData', (type) => {
   });
 });
 
+Cypress.Commands.add(
+  'loadSampleDataForWorkspace',
+  (type, workspaceId, datasourceId) => {
+    cy.request({
+      method: 'POST',
+      headers: { 'osd-xsrf': 'opensearch-dashboards' },
```
[the remark this answers is not in the document]
Comment (Hailong-am) [reply cypress/utils/commands.js:603]: There is no difference between `true` and other values. as long as `osd-xsrf` exists in http header.

Reference: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/http/lifecycle_handlers.ts#L55-L59

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_analytics_overviews.spec.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      cy.go('back');
+
+      // view all
+      cy.contains('View all').click();
+      // verify url has /app/objects
+      cy.url().should('include', 'app/objects');
+    });
+
+    it('should display Recent alerts panel', () => {
+      cy.contains('Recent alerts').should('be.visible');
```
[the remark this answers is not in the document]
Comment (Hailong-am) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_analytics_overviews.spec.js:120]: You are right, i am going remove both alerts/ threat alerts card related test from this test spec.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_search_overviews.spec.js (hunk 1 above)
Comment (Hailong-am) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_search_overviews.spec.js:35]: `cy.createWorkspace` already been a util function, and this `createWorkspace` function just utilized the `cy.createWorkspace` to create workspace with and without datasource.  To move it to a common place as util function may not benefits other test cases.

**Comment by wanglam** [comment]
@Hailong-am The `workspace_analytics_overviews` failed on workspace test workflow. Could you help to take a look?

**Review by SuZhou-Joe — APPROVED** [review]

**Review by wanglam — APPROVED** [review]

**Pull request merged**